packages/db-orchestrator/index.cjs

```diff
@@ -2,33 +2,48 @@ const { readdirSync, readJSONSync, writeJSONSync, pathExistsSync, emptyDirSync,
 const md5 = require("blueimp-md5")
 const chalk = require('chalk')
 const logEvent = require('@evidence-dev/telemetry')
+//const getColumnSummary = require('@evidence-dev/components/modules/getColumnSummary');
 const readline = require('readline');
 
-const getCache = function (dev, queryString, queryTime, settings) {
-    queryTime = md5(queryTime)
-    if (dev) {
-        const cache = readJSONSync("./.evidence-queries/cache/" + queryTime + "/" + md5(queryString) + ".json", { throws: false })
-        if (cache) {
-            logEvent("cache-query", dev, settings)
-            return cache
-        }
+const cacheDirectory = "./.evidence-queries/cache";
+
+var EvidenceType;
+(function (EvidenceType) {
+    EvidenceType["BOOLEAN"] = "boolean";
+    EvidenceType["NUMBER"] = "number";
+    EvidenceType["STRING"] = "string";
+    EvidenceType["DATE"] = "date";
+})(EvidenceType || (EvidenceType = {}));
+
+var TypeFidelity;
+(function (TypeFidelity) {
+    TypeFidelity["INFERRED"] = "inferred";
+    TypeFidelity["PRECISE"] = "precise";
+})(TypeFidelity || (TypeFidelity = {}));
+
+
+const getQueryCachePaths = (queryString, queryTime) => {
+    let queryTimeMD5 = md5(queryTime);
+    let queryStringMD5 = md5(queryString);
+    let path = `${cacheDirectory}/${queryTimeMD5}`;
+    return {
+        'cacheDirectory': path,
+        'resultsCacheFile': `${cacheDirectory}/${queryTimeMD5}/${queryStringMD5}.json`,
+        'columnTypeCacheFile': `${cacheDirectory}/${queryTimeMD5}/${queryStringMD5}-column-types.json`,
     }
 }
 
-const updateCache = function (dev, queryString, data, queryTime) {
-    queryTime = md5(queryTime)
-    if (dev) {
-        if (!pathExistsSync("./.evidence-queries")) {
-            mkdirSync("./.evidence-queries")
+const updateCache = function (devMode, queryString, data, columnTypes, queryTime) {
+    if (devMode) {
+        const {cacheDirectory, resultsCacheFile, columnTypeCacheFile} = getQueryCachePaths(queryString, queryTime);
+        if (!pathExistsSync(cacheDirectory)) {
+            emptyDirSync(cacheDirectory); //TODO I think this invalidates the entire cache.  Check if this is by design.
+            mkdirSync(cacheDirectory, { recursive: true });
         }
```

Comment:
RE comment ^^^.  I am pretty sure this is by design.

Reply:
Yes, this is by design, the cacheing strategy is pretty rudimentary right now. The name of the directory it's looking for is based on truncating the current time to the hour. If it does exist, then we are within the hour and we can add to the existing cache. If it doesn't exist, then whatever might exist is out of date and we invalidate it all, replacing it with our fresh directory for the hour. 

We'll want to move "clear cache and refresh" into the dev UI soon enough.